src/Nethermind/Nethermind.Blockchain/Receipts/ReceiptsIterator.cs

```diff
@@ -42,7 +61,15 @@ public bool TryGetNext(out TxReceiptStructRef current)
             {
                 if (_decoderContext.Position < _length)
                 {
-                    ReceiptStorageDecoder.Instance.DecodeStructRef(ref _decoderContext, RlpBehaviors.Storage, out current);
+                    if (_compactEncoding)
+                    {
+                        CompactReceiptStorageDecoder.Instance.DecodeStructRef(ref _decoderContext, RlpBehaviors.Storage, out current);
+                    }
+                    else
+                    {
+                        ReceiptStorageDecoder.Instance.DecodeStructRef(ref _decoderContext, RlpBehaviors.Storage, out current);
+                    }
+                    _recoveryContext?.RecoverReceiptData(ref current);
```

Comment:
in non-compact encoding, we don't have to recover?

Reply:
We don't right? Its already recovered during insert.

Reply:
yes it should be. Here you are recovering always, right?

Reply:
No, recovery context will be null if its not compact encoding.